athenz-identity-provider-service/src/main/java/com/yahoo/vespa/hosted/athenz/instanceproviderservice/AthenzSslTrustStoreConfigurator.java

```diff
@@ -0,0 +1,109 @@
+        this.trustStore = createTrustStore(keyProvider, configserverConfig, athenzProviderServiceConfig);
+    }
+
+    @Override
+    public void configure(SslTrustStoreContext sslTrustStoreContext) {
+        sslTrustStoreContext.updateTrustStore(trustStore);
+        log.log(LogLevel.INFO, "Configured JDisc trust store with self-signed certificate");
+    }
+
+    private static KeyStore createTrustStore(KeyProvider keyProvider,
+                                             ConfigserverConfig configserverConfig,
+                                             AthenzProviderServiceConfig athenzProviderServiceConfig) {
+        try {
+            KeyPair keyPair = getKeyPair(keyProvider, configserverConfig, athenzProviderServiceConfig);
+            X509Certificate selfSignedCertificate = createSelfSignedCertificate(keyPair, configserverConfig);
+            log.log(LogLevel.FINE, "Generated self-signed certificate: " + selfSignedCertificate);
+            KeyStore trustStore = KeyStore.getInstance("JKS");
+            trustStore.load(null);
+            trustStore.setCertificateEntry("cfgselfsigned", selfSignedCertificate);
+            return trustStore;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static KeyPair getKeyPair(KeyProvider keyProvider,
+                                      ConfigserverConfig configserverConfig,
+                                      AthenzProviderServiceConfig athenzProviderServiceConfig) {
+        String key = configserverConfig.environment() + "." + configserverConfig.region();
+        AthenzProviderServiceConfig.Zones zoneConfig = athenzProviderServiceConfig.zones(key);
+        return keyProvider.getKeyPair(zoneConfig.secretVersion());
+    }
+
+    private static X509Certificate createSelfSignedCertificate(KeyPair keyPair, ConfigserverConfig config)
+            throws IOException, CertificateException, OperatorCreationException {
+        ContentSigner contentSigner = new JcaContentSignerBuilder("SHA256WithRSA").build(keyPair.getPrivate());
+        X500Name x500Name = new X500Name("CN="+ config.loadBalancerAddress());
+        Instant now = Instant.now();
+        Date notBefore = Date.from(now);
+        Date notAfter = Date.from(now.plus(Duration.ofDays(30)));
+
+        GeneralNames generalNames = new GeneralNames(
+                config.zookeeperserver().stream()
+                        .map(server -> new GeneralName(GeneralName.dNSName, server.hostname()))
+                        .toArray(GeneralName[]::new));
+
+        X509v3CertificateBuilder certificateBuilder =
+                new JcaX509v3CertificateBuilder(
+                        x500Name, BigInteger.valueOf(now.toEpochMilli()), notBefore, notAfter, x500Name, keyPair.getPublic()
+                )
+                        .addExtension(new ASN1ObjectIdentifier("2.5.29.19"), true, new BasicConstraints(true))
```

Comment:
Why not just `Extension.basicConstraints`?

Reply:
No good reason really. Fixed in latest commit 👍